### ISSUE TITLE
optimizer.init(error) causes exception in Windows 10 using boost 1.62.0

### ISSUE THREAD

**Issue by danielfppps** [Shark-ML/Shark#128]
When using the NN example [here](http://image.diku.dk/shark/sphinx_pages/build/html/rest_sources/tutorials/algorithms/ffnet.html) the optimizer.init(error) line causes a crash - with no discernible exception - in Windows 10 (using boost v1.62.0 or v1.61.0). The error seems to happen regardless of loss function or optimizer used (tried all possible error functions and optimizers to no avail). The same code using the same boost library seems to work fine on Linux. Has anyone experienced this same issue? 

By the way thanks a lot for developing shark, it's a really amazing and fast ML library :+1:

**Comment by Ulfgard** [comment]
Hi,

I have no windows here right now. It works on linux. 
have you tried running the full example itself? Do other examples work? if you have configured shark with examples enabled you can just build and run the target FFNNMultiClassCrossEntropy

alternatively you can copy the source at examples/Supervised/FFNNMultiClassCrossEntropy.tpp

**Comment by danielfppps** [comment]
Thanks for your reply. I can confirm that it works correctly on Linux, the problem is only when using Windows. No examples using Neural networks work for me on Windows. The programs always crash when executing the optimizer.init function. Let me know if there is any additional information I could provide. Thanks a lot for your help!

**Comment by Ulfgard** [comment]
have you tried running in debug mode? this should give some information.

**Comment by danielfppps** [comment]
I wasn't able to get any additional useful information when running in debug mode. However I tried compiling Shark in 64-bit mode and the error does not appear anymore which means that this only happens when compiling and running on Windows 32-bit mode. This means you might be able to reproduce in Linux 32-bit. I will just work with 64-bit binaries. Thanks a lot for your help!
